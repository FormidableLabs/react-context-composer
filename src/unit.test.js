/* eslint-env jest */

import ContextComposer from '.';
import React from 'react';
import TestRenderer from 'react-test-renderer';

const Theme = React.createContext('light');
const Language = React.createContext('en');

describe('composing providers', () => {
  test('empty array', () => {
    const renderer = TestRenderer.create(
      <ContextComposer contexts={[]}>
        <div />
      </ContextComposer>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
  test('[Theme]', () => {
    const renderer = TestRenderer.create(
      <ContextComposer contexts={[<Theme.Provider value="themeBar" />]}>
        <Theme.Consumer>{theme => `Theme: ${theme}`}</Theme.Consumer>
      </ContextComposer>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
  test('[Theme, Language]', () => {
    const renderer = TestRenderer.create(
      <ContextComposer
        contexts={[
          <Theme.Provider value="themeBar" />,
          <Language.Provider value="langFoo" />
        ]}
      >
        <Theme.Consumer>{theme => `Theme: ${theme}`}</Theme.Consumer>,
        <Language.Consumer>
          {language => `Language: ${language}`}
        </Language.Consumer>
      </ContextComposer>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
  test('does not mutate contexts array', () => {
    TestRenderer.create(
      <ContextComposer
        contexts={Object.freeze([
          <Theme.Provider value="themeBar" />,
          <Language.Provider value="langFoo" />
        ])}
      >
        <div />
      </ContextComposer>
    );
  });
});

describe('composing consumers', () => {
  test('empty array', () => {
    const render = jest.fn(() => null);
    const renderer = TestRenderer.create(
      <ContextComposer contexts={[]}>{render}</ContextComposer>
    );
    expect(render.mock.calls).toMatchSnapshot();
  });
  test('[Theme]', () => {
    const render = jest.fn(() => null);
    const renderer = TestRenderer.create(
      <Theme.Provider value="themeBar">
        <ContextComposer contexts={[Theme]}>{render}</ContextComposer>
      </Theme.Provider>
    );
    expect(render.mock.calls).toMatchSnapshot();
  });
  test('[Theme, Language]', () => {
    const render = jest.fn(() => null);
    const renderer = TestRenderer.create(
      <Language.Provider value="langFoo">
        <Theme.Provider value="themeBar">
          <ContextComposer contexts={[Theme, Language]}>
            {render}
          </ContextComposer>
        </Theme.Provider>
      </Language.Provider>
    );
    expect(render.mock.calls).toMatchSnapshot();
  });
  test('does not mutate contexts array', () => {
    TestRenderer.create(
      <ContextComposer contexts={Object.freeze([Theme, Language])}>
        {() => null}
      </ContextComposer>
    );
  });
});
