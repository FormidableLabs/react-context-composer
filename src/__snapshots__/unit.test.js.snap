// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composing consumers [Theme, Language] 1`] = `
Array [
  Array [
    "themeBar",
    "langFoo",
  ],
]
`;

exports[`composing consumers [Theme] 1`] = `
Array [
  Array [
    "themeBar",
  ],
]
`;

exports[`composing consumers empty array 1`] = `
Array [
  Array [],
]
`;

exports[`composing providers [Theme, Language] 1`] = `
Array [
  "Theme: themeBar",
  ",",
  "Language: langFoo",
]
`;

exports[`composing providers [Theme] 1`] = `"Theme: themeBar"`;

exports[`composing providers empty array 1`] = `<div />`;
